File: packages/shared/src/types/idea-guidance.ts
// 阶段 37：想法引导官类型定义
// 权限级别 L1（模型草案），不执行 Runner、不写文件、不改 Git

// ── Session ──
export type IdeaGuidanceSession = {
  id: string;
  project_id: string;
  status: "draft" | "questions_ready" | "seed_ready" | "cancelled" | "failed";
  source: "manual" | "model_guided";
  idea_summary: string;
  constraints_summary: string | null;
  model_call_id: string | null;
  created_by: string;
  created_at: string;
  updated_at: string;
};

// ── Question ──
export type IdeaGuidanceQuestion = {
  id: string;
  project_id: string;
  session_id: string;
  sort_order: number;
  question: string;
  answer: string | null;
  status: "pending" | "answered" | "skipped";
  created_at: string;
  updated_at: string;
};

// ── Seed ──
export type ProjectSeed = {
  id: string;
  project_id: string;
  session_id: string;
  status: "draft" | "ready" | "converted" | "cancelled";
  product_goal: string | null;
  target_users: string | null;
  mvp_scope: string | null;
  non_goals: string | null;
  key_features: string | null;
  pages_or_modules: string | null;
  data_entities: string | null;
  technical_constraints: string | null;
  acceptance_criteria: string | null;
  risk_points: string | null;
  open_questions: string | null;
  recommended_next_step: string | null;
  model_call_id: string | null;
  created_at: string;
  updated_at: string;
};

// ── Command Inputs ──
export type CreateIdeaGuidanceQuestionsInput = {
  idea: string;
  constraints?: string | null;
  model_record_id?: string | null;
  second_confirm: boolean;
  confirm_text?: string | null;
};

export type GenerateProjectSeedInput = {
  session_id: string;
  model_record_id?: string | null;
  second_confirm: boolean;
  confirm_text?: string | null;
};

export type SaveGuidanceAnswersInput = {
  session_id: string;
  answers: { question_id: string; answer: string }[];
};

// ── Command Responses ──
export type CreateIdeaGuidanceQuestionsResponse = {
  session: IdeaGuidanceSession;
  questions: IdeaGuidanceQuestion[];
  audit_record_id: string | null;
  warnings: string[];
};

export type GenerateProjectSeedResponse = {
  seed: ProjectSeed;
  session: IdeaGuidanceSession;
  audit_record_id: string | null;
  warnings: string[];
};

File: apps/desktop/src-tauri/src/commands/idea_guidance.rs
// 阶段 37：想法引导官 Tauri 命令
// 权限级别 L1（模型草案），不执行 Runner、不写文件、不改 Git。
// L1 不是免确认：真实模型调用前必须二次确认。
// 所有输入结构体使用 deny_unknown_fields 防止前端传入自由字段。

use serde::Deserialize;

use crate::db::DbState;
use crate::services::idea_guidance::{
    self, CreateIdeaGuidanceQuestionsResponse, GenerateProjectSeedResponse, IdeaGuidanceSession,
    ProjectSeed, QuestionAnswer,
};

/// 创建想法引导追问
/// 真实模型调用必须二次确认。
#[derive(Deserialize)]
#[serde(deny_unknown_fields)]
pub struct CreateIdeaGuidanceQuestionsInput {
    pub idea: String,
    #[serde(default)]
    pub constraints: Option<String>,
    #[serde(default)]
    pub model_record_id: Option<String>,
    #[serde(default)]
    pub second_confirm: bool,
    #[serde(default)]
    pub confirm_text: Option<String>,
}

#[tauri::command]
pub fn create_idea_guidance_questions(
    state: tauri::State<'_, DbState>,
    input: CreateIdeaGuidanceQuestionsInput,
) -> Result<CreateIdeaGuidanceQuestionsResponse, String> {
    let connection = state.connection()?;
    idea_guidance::create_idea_guidance_questions(
        &input.idea,
        &input.constraints,
        &connection,
        input.model_record_id.as_deref(),
        input.second_confirm,
        &input.confirm_text,
    )
}

/// 生成项目种子
/// 从 DB 读取 session 和已回答的问题，构造 prompt 调用模型。
/// 真实模型调用必须二次确认。
#[derive(Deserialize)]
#[serde(deny_unknown_fields)]
pub struct GenerateProjectSeedInput {
    pub session_id: String,
    #[serde(default)]
    pub model_record_id: Option<String>,
    #[serde(default)]
    pub second_confirm: bool,
    #[serde(default)]
    pub confirm_text: Option<String>,
}

#[tauri::command]
pub fn generate_project_seed(
    state: tauri::State<'_, DbState>,
    input: GenerateProjectSeedInput,
) -> Result<GenerateProjectSeedResponse, String> {
    let connection = state.connection()?;
    idea_guidance::generate_project_seed(
        &input.session_id,
        &connection,
        input.model_record_id.as_deref(),
        input.second_confirm,
        &input.confirm_text,
    )
}

/// 保存用户对追问的回答（不调用模型，无需二次确认）
#[derive(Deserialize)]
#[serde(deny_unknown_fields)]
pub struct SaveGuidanceAnswersInput {
    pub session_id: String,
    pub answers: Vec<QuestionAnswerInput>,
}

#[derive(Deserialize)]
#[serde(deny_unknown_fields)]
pub struct QuestionAnswerInput {
    pub question_id: String,
    pub answer: String,
}

#[tauri::command]
pub fn save_guidance_answers(
    state: tauri::State<'_, DbState>,
    input: SaveGuidanceAnswersInput,
) -> Result<IdeaGuidanceSession, String> {
    let connection = state.connection()?;
    let answers: Vec<QuestionAnswer> = input
        .answers
        .into_iter()
        .map(|a| QuestionAnswer {
            question_id: a.question_id,
            answer: a.answer,
        })
        .collect();
    idea_guidance::save_guidance_answers(&connection, &input.session_id, &answers)
}

/// 列出当前项目的所有种子
#[tauri::command]
pub fn list_project_seeds(state: tauri::State<'_, DbState>) -> Result<Vec<ProjectSeed>, String> {
    let connection = state.connection()?;
    idea_guidance::list_project_seeds(&connection)
}
